packages/swapper/src/swappers/ThorchainSwapper/endpoints.ts

```diff
@@ -1,161 +1,29 @@
-
-    const { data, to } = await getEvmData({ config, step, tradeQuote })
-
-    const adapter = assertGetEvmChainAdapter(sellAsset.chainId)
-
-    const value = isNativeEvmAsset(sellAsset.assetId)
-      ? sellAmountIncludingProtocolFeesCryptoBaseUnit
-      : '0'
-
-    const feeData = await evm.getFees({ adapter, data, to, value, from, supportsEIP1559 })
-
-    return feeData.networkFeeCryptoBaseUnit
-  },
-  getUnsignedUtxoTransaction: async ({
-    tradeQuote,
-    stepIndex,
-    xpub,
-    accountType,
-    assertGetUtxoChainAdapter,
-    config,
-  }) => {
-    if (!isExecutableTradeQuote(tradeQuote)) throw new Error('Unable to execute a trade rate quote')
-
-    const { memo } = tradeQuote as ThorTradeQuote
-    if (!memo) throw new Error('Memo is required')
-
-    const step = getExecutableTradeStep(tradeQuote, stepIndex)
-
-    const { accountNumber, sellAmountIncludingProtocolFeesCryptoBaseUnit, sellAsset, feeData } =
-      step
-
-    const { vault, opReturnData } = await getUtxoThorTxInfo({
-      sellAsset,
-      xpub,
-      memo,
-      config,
-    })
-
-    return assertGetUtxoChainAdapter(sellAsset.chainId).buildSendApiTransaction({
-      value: sellAmountIncludingProtocolFeesCryptoBaseUnit,
-      xpub,
-      to: vault,
-      accountNumber,
-      // skip address validation for thorchain vault addresses as they may exceed the risk score threshold, but are still valid for use
-      skipToAddressValidation: true,
-      chainSpecific: {
-        accountType,
-        opReturnData,
-        satoshiPerByte: (feeData.chainSpecific as UtxoFeeData).satsPerByte,
-      },
-    })
-  },
-  getUtxoTransactionFees: async ({
-    stepIndex,
-    tradeQuote,
-    xpub,
-    assertGetUtxoChainAdapter,
-    config,
-  }) => {
-    if (!isExecutableTradeQuote(tradeQuote)) throw new Error('Unable to execute a trade rate quote')
-
-    const { memo } = tradeQuote as ThorTradeQuote
-    if (!memo) throw new Error('Memo is required')
-
-    const step = getExecutableTradeStep(tradeQuote, stepIndex)
-
-    const { sellAsset, sellAmountIncludingProtocolFeesCryptoBaseUnit } = step
-
-    const adapter = assertGetUtxoChainAdapter(sellAsset.chainId)
-
-    const { vault, opReturnData } = await getUtxoThorTxInfo({
-      sellAsset,
-      xpub,
-      memo,
-      config,
-    })
-
-    const { fast } = await adapter.getFeeData({
-      to: vault,
-      value: sellAmountIncludingProtocolFeesCryptoBaseUnit,
-      chainSpecific: { pubkey: xpub, opReturnData },
-      sendMax: false,
-    })
-
-    return fast.txFee
-  },
+  getTradeRate,
+  getTradeQuote,
+  getUnsignedEvmTransaction: input => evm.getUnsignedEvmTransaction(input, swapperName),
+  getEvmTransactionFees: input => evm.getEvmTransactionFees(input, swapperName),
+  getUnsignedUtxoTransaction: input => utxo.getUnsignedUtxoTransaction(input, swapperName),
+  getUtxoTransactionFees: input => utxo.getUtxoTransactionFees(input, swapperName),
```

Comment:
nitpick: don't use evm and utxo prefix here as this looks generalised across swapper and not THOR specific